dotnet/src/webdriver/DevTools/DevToolsSession.cs

```diff
@@ -600,7 +600,19 @@ private void ProcessMessage(string message)
                 var methodParts = method.Split(new char[] { '.' }, 2);
                 var eventData = messageObject["params"];
 
-                LogTrace("Recieved Event {0}: {1}", method, eventData.ToString());
```

Comment:
This is not a complete solution, the JSON exception is simply thrown later: 
![image](https://github.com/user-attachments/assets/54d22891-71b8-42b0-a39d-af60929b3c7c)

The above is in generated cdp code (`\cdp\v131\Fetch\FetchAdapter.cs` in the above screenshot).

Related relevant code:
![image](https://github.com/user-attachments/assets/441f0cbc-41fd-493f-8043-1d231eef0d7a)

Reply:
@nvborisenko You mentioned trying out `JavaScriptEncoder.UnsafeRelaxedJsonEscaping`. I cannot edit the generated code directly but I added a spoof here, and it did not fix the exception
![image](https://github.com/user-attachments/assets/3530df4f-a53a-43af-9e6e-1d901b289c80)

Reply:
I modified generator to use `JavaScriptEncoder.UnsafeRelaxedJsonEscaping`, it didn't help. The worst scenario is that it works with `Newtonsoft.Json` (Selenium v4.23?).

Reply:
I do not understand how Newtonsoft accepts this nonsense input. It’s not real JSON, just some raw binary data.

Maybe we should intersept the post data and sanitize it ourselves?

Reply:
One more observation: in json response I see binary data as is, probably not even encoded! This is why VS Code shows json improperly. CDP specification doesn't say it should be encoded (base64). I think this is why they deprecated it.

PS: But why `Newtonsoft.Json` could parse it successfully? I think it is issue in this library. I didn't check it yet, but I guess:

1 using old selenium version (v4.23?) when we upload binary file
2 and we download it manually back
3 then the content of "original" and "downloaded" files is different

Reply:
I tried switching from `JsonNode` to `JsonElement` just to see what would happen.

Good news, no exceptions on `ToString()` like the logging has, but the de-serialization still fails
![image](https://github.com/user-attachments/assets/6d22e3cb-1bc4-4a0a-a542-da852f8ec31e)

Maybe we should parse and encode the postData ourselves?

Reply:
I am stupid for this activity. It is easier to:
- really `Newtonsoft.Json` parses it correctly?

Reply:
I narrowed down the issue. It looks like this character [`U+DB40`](https://unicodeplus.com/U+DB40) is appearing here for some reason. It is the first part of a surrogate pair (representing a character using two UNICODE characters instead of one) but the second half is missing.
![image](https://github.com/user-attachments/assets/b65d8b2d-9210-4284-a455-202d8c8eaed7)

I opened the raw `Firefox Installer.exe` file and found this segment here:
![image](https://github.com/user-attachments/assets/b337e747-67ef-4238-a345-79ca18488813)

Note how in this case, it is just a regular [`I`](https://unicodeplus.com/U+0049).

What is happening here? The CDP is giving us back broken JSON, this may be a bug on their end.

I will investigate whether it is possible to work around this somehow.

Reply:
I found a workaround, with a custom converter.

We can add to the postData property:

![image](https://github.com/user-attachments/assets/6a5630c0-1aea-4ea5-828e-1534b6959f17)

As well as to PostDataEntry.Bytes:
![image](https://github.com/user-attachments/assets/fc4e54e4-c4bf-4349-8180-11728c5ba595)


The following converter:

```csharp
internal sealed class InvalidUtf16Converter : JsonConverter<string>
{
    public override bool HandleNull => true;

    public override string? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        try
        {
            return reader.GetString();
        }
        catch (InvalidOperationException)
        {
            var bytes = reader.ValueSpan;
            var sb = new StringBuilder(bytes.Length);
            foreach (var b in bytes)
                sb.Append(Convert.ToChar(b));
            return sb.ToString();
        }
    }

    public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options) =>
        writer.WriteStringValue(value);
}
```

Since we have a good idea of which models may have arbitrary data, 

However... it only works with `JsonElement.Deserialize<T>`. When using a `JsonNode.Deserialize<T>`, the exception happens way before even the converter is reached. Even `JsonNode.ToString()` throws an exception!

For this reason, we may need to change various `JsonNode`s to `JsonElement`s, such as `DevToolsEventReceivedEventArgs.EventData` (that property is used everywhere in the cdp-generated code, but luckily the line `e.EventData.Deserialize(eventData.EventArgsType)` works exactly the same (it just hits a different overload).

I am not familiar enough with the cdp code generation process to make generator changes. Is there any docs for this?